src/components/visualizations/orbitalViewer/PlaybackSpeed.jsx

```diff
@@ -24,79 +25,53 @@ function PlaybackSpeed({
   sliderOnChangeCallback,
   speeds,
 }) {
+  const { t } = useTranslation('widgets');
+
   function formatSpeed(speed) {
     const years = speed / 365;
     const justYears = Math.floor(years);
     const justDays = 365 * (years - justYears);
     let formattedYears = {};
     let formattedDays = {};
 
-    if (justYears >= 1) {
-      formattedYears = {
-        number: formatValue(justYears),
-        string: `Year${justYears >= 2 ? 's' : ''}`,
-      };
-    } else {
-      formattedYears = {
-        number: 0,
-        string: 'Years',
-      };
-    }
+    formattedYears = {
+      number: formatValue(justYears),
+      string: t('orbit_viewer.playback.interval.year', { count: justYears }),
+    };
 
-    if (justDays > 1) {
-      formattedDays = {
-        number: formatValue(justDays),
-        string: `Day${justDays >= 2 ? 's' : ''}`,
-      };
-    } else {
-      formattedDays = {
-        number: 0,
-        string: 'Days',
-      };
-    }
+    formattedDays = {
+      number: formatValue(justDays),
+      string: t('orbit_viewer.playback.interval.day', { count: justDays }),
+    };
 
     if (justYears >= 1) {
       return `${formattedYears.number} ${formattedYears.string}`;
     }
 
     if (justDays < 1) {
-      return `1 sec`;
+      return t('orbit_viewer.playback.interval.secWithCount', { count: 1 });
     }
 
     return `${formattedDays.number} ${formattedDays.string}`;
   }
 
   function formatElapsed(type, value) {
-    const sign = elapsedTime < 0 ? '-' : '';
+    const isNegative = elapsedTime < 0;
     const years = Math.abs(elapsedTime) / 365.256;
     const justYears = Math.floor(years);
-    const justDays = 365.256 * (years - justYears);
+    const justDays = Math.floor(365.256 * (years - justYears));
```

Comment:
needed to floor days because it was being stored as a float which when passed to i18next would render the time interval as a plural (i.e. 1.2 days instead of 1 day)